gecko_client/pchain_api.go

```diff
@@ -13,25 +13,25 @@ type PChainApi struct {
 	rpcRequester geckoJsonRpcRequester
 }
 
-type Validator struct {
-	StartTime string `json:"startTime"`
-	EndTime string	`json:"endTime"`
-	StakeAmount string	`json:"stakeAmount"`
-	Id string	`json:"id"`
-}
-
-type ValidatorList struct {
-	Validators []Validator	`json:"validators"`
-}
+func (api PChainApi) GetBlockchainStatus(blockchainId string) (BlockchainStatus, error) {
+	params := map[string]interface{}{
+		"blockchainID": blockchainId,
+	}
+	responseBodyBytes, err := api.rpcRequester.makeRpcRequest(pchainEndpoint, "platform.getBlockchainStatus", params)
+	if err != nil {
+		return BlockchainStatus{}, stacktrace.Propagate(err, "Error making request")
+	}
 
-type GetValidatorsResponse struct {
-	JsonRpcVersion string	`json:"jsonrpc"`
-	Result ValidatorList	`json:"result"`
-	Id int	`json:"id"`
+	// TODO try moving this inside the MakeRequest method, even though Go doesn't have generics
+	var response GetBlockchainStatusResponse
+	if err := json.Unmarshal(responseBodyBytes, &response); err != nil {
+		return BlockchainStatus{}, stacktrace.Propagate(err, "Error unmarshalling JSON response")
+	}
+	return response.Result, nil
 }
 
 func (api PChainApi) GetCurrentValidators() ([]Validator, error) {
-	responseBodyBytes, err := api.rpcRequester.makeRpcRequest(pchainEndpoint, "platform.getCurrentValidators")
+	responseBodyBytes, err := api.rpcRequester.makeRpcRequest(pchainEndpoint, "platform.getCurrentValidators", make(map[string]interface{}))
```

Comment:
I think we need to make this change in the admin API as well, travisCI failing with:

The command "eval go get -v -t ./... " failed. Retrying, 2 of 3.
github.com/kurtosis-tech/ava-e2e-tests/gecko_client
github.com/kurtosis-tech/ava-e2e-tests/gecko_client
gecko_client/admin_api.go:60:59: not enough arguments in call to api.rpcRequester.makeRpcRequest
	have (string, string)
	want (string, string, map[string]interface {})
The command "eval go get -v -t ./... " failed. Retrying, 3 of 3.
github.com/kurtosis-tech/ava-e2e-tests/gecko_client
github.com/kurtosis-tech/ava-e2e-tests/gecko_client
gecko_client/admin_api.go:60:59: not enough arguments in call to api.rpcRequester.makeRpcRequest
	have (string, string)
	want (string, string, map[string]interface {})